### PR TITLE
REF: move most of Block.interpolate to an array method

### PR DESCRIPTION
**Pull request by jbrockmendel**
Not _quite_ at the point where it can be called directly from ArrayManager, but close.

### CODE REVIEW

File: pandas/core/missing.py [hunk 1]
```diff
@@ -99,6 +103,18 @@ def mask_missing(arr: ArrayLike, values_to_mask) -> np.ndarray:
     return mask
 
 
+def try_clean_fill_method(method):
```
Comment (jreback) [inline pandas/core/missing.py:106]: this is a bitty smelly, can you just use clean_fill_method?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/missing.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/missing.py:106]: sure

Comment (jbrockmendel) [reply pandas/core/missing.py]: updated + greenish

**Comment by jbrockmendel** [comment]
@jorisvandenbossche have you gotten a chance to look at the Categorical-ArrayManager failure on master?

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
any issues with AM tests @jbrockmendel ?

**Comment by jbrockmendel** [comment]
> any issues with AM tests

nothing showing up in the CI.  AM doesnt (yet) call this directly

**Comment by jorisvandenbossche** [comment]
> Not quite at the point where it can be called directly from ArrayManager, but close.

What's needed additionally to have it usable for ArrayManager?

**Comment by jbrockmendel** [comment]
> What's needed additionally to have it usable for ArrayManager?

the short-circuiting on L1113 and L1123 need some special attention (particularly the one on 1123).  Then maybe_downcast at the end needs porting (ive got a branch but am not quite happy with it yet)

**Comment by jreback** [comment]
can you merge master and ping on green

**Comment by jbrockmendel** [comment]
ping

**Pull request merged**